### PR TITLE
[pack] adding more properties to Azure Monitor logs

### LINKED ISSUE #6556: Improvements to FunctionAppLogs
We need to expand the details we send to Azure Monitor Diagnotsic Logs (the FunctionAppLogs). Add the following properties:
1. AppName
2. MachineName (need to figure out what best name is here)
3. LevelId (for better filtering of, for example `LevelId > 2`
4. EventName
5. EventId

### PR DESCRIPTION
**Pull request by brettsam**
Adding new properties to FunctionAppLogs that will allow customers to run more interesting queries in Log Analytics.

resolves #6556 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [x] Otherwise: Backport tracked by issue/PR #6679
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by brettsam** [comment]
@mathewc / @paulbatum -- I think you guys may have a good understanding of all our names here so looking for advice. 

We currently have the following app names that we use in our backend logs:
1. HostName -- from `WEBSITE_HOSTNAME` or `WEBSITE_SITE_NAME`. I believe this **does change** during a slot swap, which is why we Synchronize it.
2. AppName -- combines `WEBSITE_SITE_NAME` with `WEBSITE_SLOT_NAME` (if it's not "production"). I *think* that this **does not change** during a slot swap (which is why HostName could be better?)
3. RuntimeSiteName -- from `WEBSITE_DEPLOYMENT_ID`

What do you think is the most useful to customers? I'm currently using AppName but it seems like HostName may be what people would be after.

**Comment by paulbatum** [comment]
@brettsam if hostname picks up any custom hostname settings, then I think it should be a separate information element, rather than a replacement for appname.

**Comment by brettsam** [comment]
Just coming back to this -- host name (@mathewc can correct me) is the the default host name, I believe -- no custom host names there. But for something like a staging slot it'd show up there with `-staging`, etc.

What about RuntimeSiteName? I assume we don't want to expose that publicly as it'd be confusing during slot swaps?

I'm trying to think through what's the most useful while not being confusing.

I'm thinking of using the host name as the "app name" (as long as it doesn't include custom names) because that does update after a swap. So if you're trying to get logs from your prod slot it's really the only thing we update during a swap.

File: src/WebJobs.Script/Config/AppServiceOptionsSetup.cs [hunk 1]
```diff
@@ -20,6 +20,22 @@ public void Configure(AppServiceOptions options)
...
+            if (string.IsNullOrEmpty(instanceName))
+            {
+                instanceName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ComputerName);
+                if (string.IsNullOrEmpty(instanceName))
+                {
+                    instanceName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
+                }
+            }
+
+            return instanceName;
```
Comment (brettsam) [inline src/WebJobs.Script/Config/AppServiceOptionsSetup.cs]: @balag0 -- is this still the best way to identify an instance name in Linux? I got this from your PR SDK but wanted to double-check: https://github.com/Azure/azure-webjobs-sdk/pull/2240

**Comment by paulbatum** [comment]
I'm not sure I understand why this decision is specific for the azure monitor logs. Isn't there prior art here in terms of what we use for the appinsights logging?

File: src/WebJobs.Script/Config/AppServiceOptionsSetup.cs (hunk 1 above)
Comment (balag0) [reply src/WebJobs.Script/Config/AppServiceOptionsSetup.cs]: Yes, In AppService AzureWebsiteInstanceId will be unique per container instance and in consumption ContainerName will be unique.

**Comment by brettsam** [comment]
I was revisiting how we do this today b/c I thought that the env vars didn't get reset during a swap (which is why we've got the host name header synchronizing stuff).  But I just realized that a swap always restarts the app... 

Basically, ignore all my confusing questions :-)

**Comment by paulbatum** [comment]
Yes a swap does restart an app, but I think the runtime looks for the hostname change separately because that is something that happens after the restart. For example, the app restarts when the config from prod is pushed to the code that is in stage. Then once that change looks good, then the hostname updates. So the only case where logging the hostname in addition to app name would be useful is if you want to be able to identify those two stages in your logs. .

**Comment by brettsam** [comment]
+@yojagad since he just did something similar in #6653

File: src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs [hunk 2]
```diff
@@ -93,6 +104,8 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             using (JsonTextWriter writer = new JsonTextWriter(sw) { Formatting = Formatting.None })
             {
                 writer.WriteStartObject();
+                WritePropertyIfNotNull(writer, "appName", _appServiceOptions.CurrentValue.AppName);
```
Comment (yojagad) [inline src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs:107]: >_appServiceOptions.CurrentValue.AppName [](start = 58, length = 39)

Just double checking. The appName is different before and after specialization, right ? If yes, this does get refreshed after specialization ?

Comment (brettsam) [reply src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs:107]: Yes, it's registered with the Specialization change token: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L146

**Review by yojagad — APPROVED** [review]

**Comment by brettsam** [comment]
Just took a look in a private site extension and logs look as expected.

**Pull request merged**